File: codes/Garnetwzy/55.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int dis = 0;
        for(int i = 0; i <= dis; i++) {
            dis = max(i + nums[i], dis);
            if(dis >= nums.size()-1)
                return true;
        }
        return false;
    }
};

File: codes/HellDream/55.java
class Solution {
    public boolean canJump(int[] nums) {
        if(nums.length<=1) return true;
        int max = 0;
        for(int i=0;i<nums.length&&i<=max;i++){
            if(nums[i]!=0)
                max = Math.max(max, nums[i]+i);
            if(max>=nums.length-1) return true;
        }
        return false;
    }
}